.github/workflows/build.yml

```diff
@@ -463,10 +463,12 @@ jobs:
         uses: dtolnay/rust-toolchain@stable
         with:
           components: clippy
-      # - we want to be free of any warnings, so deny them
-      # - disable incompatible_msrv as it does not understand that we apply our
-      #   MSRV to the just the core crate.
-      - run: ./admin/clippy -- --deny warnings --allow clippy::incompatible_msrv
+      # We want to be free of any warnings, so deny them.
+      # - Allow `clippy::disallowed_types` as a workaround since `disallowed_types` configured in `.clippy.toml`
+      #   is only intended for main `rustls` crate.
+      - run: ./admin/clippy -- --deny warnings --allow clippy::disallowed_types
+      # - keep the main crate free of all Clippy warnings
+      - run: cargo clippy -p rustls -- --deny warnings
```

Comment:
I think we need to finish review & hopeful merging of other PR #2310 first, to keep that update separate.